projects/microphysics/experiments/rnn-optimize/argo_gpu.yaml

```diff
@@ -0,0 +1,139 @@
+  arguments:
+    parameters:
+    - name: tag
+      value: ":8042f5711237de53e2c539ddc11ae84ec8d795e2"
+    - name: wandb-run-group
+      value: ""
+    - name: training-config
+    - name: flags
+      value: ""
+  volumes:
+    - name: gcp-key-secret
+      secret:
+        defaultMode: 420
+        secretName: gcp-key
+  templates:
+    - name: training
+      steps:
+      - - name: train-model
+          template: train-model
+          continueOn:
+            failed: true
+          arguments:
+            parameters:
+            - name: training-config
+              value: "{{workflow.parameters.training-config}}"
+            - name: flags
+              value: "{{workflow.parameters.flags}}"
+      - - name: score-model
+          template: score-model
+          arguments:
+            parameters:
+            - name: training-config
+              value: "{{workflow.parameters.training-config}}"
+            - name: flags
+              value: "{{workflow.parameters.flags}}"
+    - name: train-model
+      inputs:
+        parameters:
+          - name: training-config
+          - {name: flags, value: " "}
+      container:
+        image: "us.gcr.io/vcm-ml/prognostic_run{{workflow.parameters.tag}}"
+        command: ["bash", "-c", "-x"]
+        envFrom:
+        - secretRef:
+            name: wandb-andrep
+        env:
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: /secret/gcp-credentials/key.json
+          - name: CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE
+            value: /secret/gcp-credentials/key.json
+          - name: WANDB_NAME
+            value: "{{workflow.name}}"
+          - name: WANDB_RUN_GROUP
+            value: "{{workflow.parameters.wandb-run-group}}"
+        volumeMounts:
+          - mountPath: /secret/gcp-credentials
+            name: gcp-key-secret
+        resources:
+          limits:
+            cpu: "7"
+            memory: "20Gi"
+            nvidia.com/gpu: 1
+          requests:
+            cpu: "4"
+            memory: "15Gi"
+        args:
+          - |
+            echo "{{inputs.parameters.training-config}}" | base64 --decode > training_config.yaml
+            export LD_LIBRARY_PATH=/usr/local/cuda/lib64:/usr/local/nvidia/lib64:$LD_LIBRARY_PATH
+            export PATH=/usr/local/nvidia/bin:$PATH
+            nvidia-smi
+
+            python3 -m fv3fit.train_microphysics \
+              --config-path training_config.yaml \
+              --wandb.job_type train \
+              {{inputs.parameters.flags}}
+      tolerations:
+      - key: "dedicated"
+        operator: "Equal"
+        value: "gpu-sim-pool"
+        effect: "NoSchedule"
+      - key: "nvidia.com/gpu"
+        operator: "Exists"
+        effect: "NoSchedule"
+    - name: score-model
+      inputs:
+        parameters:
+          - name: training-config
+          - {name: flags, value: " "}
+      container:
+        image: "us.gcr.io/vcm-ml/prognostic_run{{workflow.parameters.tag}}"
```

Comment:
Should have some conditional GPU logic and merge this into the training argo (or add a new template to the training argo)